### PR TITLE
Updates REST API 21-05-12

### PR DESCRIPTION
**Pull request by augustak**

### CODE REVIEW

File: src/main/java/ai/lucidtech/las/sdk/Field.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    public Field setDescription(String description) {
+        this.description = description;
+        return this;
+    }
+
+    public JSONObject addOptions(JSONObject body) {
+        if (this.fieldType != null) {
+            this.addOption(body, "type", this.fieldType.value);
+        }
```
Comment (magnurud) [inline src/main/java/ai/lucidtech/las/sdk/Field.java:44]: Does this work? "type" is meant to be a required field, that is either a string or null

File: src/main/java/ai/lucidtech/las/sdk/Field.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        this.addOption(body, "maxLength", this.maxLength);
+        this.addOption(body, "description", this.description == null ? "" : this.description); // TODO: Remove hack
```
Comment (magnurud) [inline src/main/java/ai/lucidtech/las/sdk/Field.java:46]: So the user is not allowed to set the description to null? This seems unfortunate.

File: src/main/java/ai/lucidtech/las/sdk/UpdateModelOptions.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    public UpdateModelOptions setPreprocessConfig(PreprocessConfig preprocessConfig) {
+        this.preprocessConfig = preprocessConfig;
+        return this;
+    }
+
+    public JSONObject addOptions(JSONObject body) {
+        this.addOption(body, "width", this.width);
+        this.addOption(body, "height", this.height);
+        this.addOption(body, "fieldConfig", this.fieldConfig);
```
Comment (magnurud) [inline src/main/java/ai/lucidtech/las/sdk/UpdateModelOptions.java:47]: These are optional, so it seems to me like you need some if-sentences here.

File: src/main/java/ai/lucidtech/las/sdk/UpdateModelOptions.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        if (this.modelStatus != null) {
+            this.addOption(body, "status", this.modelStatus.value);
+        }
+        this.addOption(body, "preprocessConfig", this.preprocessConfig);
```
Comment (magnurud) [inline src/main/java/ai/lucidtech/las/sdk/UpdateModelOptions.java:51]: also an optional argument.

File: src/test/java/lucidtech/test/ModelsTest.java [hunk 5]
```diff
@@ -32,6 +32,79 @@ private void assertModels(JSONObject models) throws IOException {
...
+            .setMaxPages(3);
+
+        CreateModelOptions options = new CreateModelOptions()
+            .setName("Model Name")
+            .setDescription("Model Description")
+            .setPreprocessConfig(preprocessConfig);
+
+        int width = 800;
+        int height = 800;
+        FieldConfig fieldConfig = new FieldConfig()
```
Comment (magnurud) [inline src/test/java/lucidtech/test/ModelsTest.java:62]: I suggest adding descriptions as well here.

**Review by magnurud — CHANGES_REQUESTED** [review]

File: src/main/java/ai/lucidtech/las/sdk/UpdateModelOptions.java (hunk 3 above)
Comment (augustak) [reply src/main/java/ai/lucidtech/las/sdk/UpdateModelOptions.java:47]: This is handled in the this.addOption method

File: src/main/java/ai/lucidtech/las/sdk/UpdateModelOptions.java (hunk 4 above)
Comment (augustak) [reply src/main/java/ai/lucidtech/las/sdk/UpdateModelOptions.java:51]: This is handled in the this.addOption method

**Review by magnurud — APPROVED** [review]

**Pull request merged**